play-json/shared/src/main/scala/Writes.scala

```diff
@@ -301,10 +301,16 @@ sealed trait LowPriorityWrites extends EnvWrites {
   /**
    * Serializer for Traversables types.
    */
-  implicit def traversableWrites[A: Writes] = {
+  implicit def traversableWrites[A: Writes]: Writes[Traversable[A]] = {
     val w = implicitly[Writes[A]]
 
-    Writes[Traversable[A]] { as => JsArray(as.map(w.writes(_)).toArray[JsValue]) }
+    Writes[Traversable[A]] { as =>
+      val builder = mutable.ArrayBuilder.make[JsValue]()
+      for (a <- as) {
```

Comment:
`as.foreach`? As each iteration is side-effect

Reply:
This doesn't have a `yield` so it is just side effecting, and desugars to `foreach`. That said, there is is no real advantage in readability here for `for` vs `foreach` so I'll switch to foreach.